### PR TITLE
Misc changes/cleanups

### PR DESCRIPTION
**Pull request by naoyam**
Some miscellaneous changes that are extracted from #1772.

### CODE REVIEW

File: third_party/nvfuser/csrc/codegen.cpp [hunk 1]
```diff
@@ -2156,6 +2156,8 @@ class CudaKernelGenerator : private OptOutConstDispatch {
         grouped_gwop->sync_buffer()->buffer()->as<TensorView>();
     func_args.arg("&").append(ir_utils::varName(sync_buffer)).append("[0]");
 
+    addProfileArguments(func_args, grouped_gwop);
```
Comment (naoyam) [inline third_party/nvfuser/csrc/codegen.cpp:2159]: Was missing.

File: third_party/nvfuser/csrc/grouped_reduction.cpp [hunk 2]
```diff
@@ -35,9 +46,10 @@ bool hasMatchingTransformations(TensorView* ref, TensorView* other) {
 }
 
 // Validate grouping of reductions and return a new max producer position
-void validateReductionGrouping(
+bool validateReductionGrouping(
```
Comment (naoyam) [inline third_party/nvfuser/csrc/grouped_reduction.cpp:49]: Changed the validation to not fail unless instructed. Intended to be used by schedulers.

File: third_party/nvfuser/csrc/type.h [hunk 3]
```diff
@@ -430,7 +430,7 @@ const char* const kMagicZeroName = "nvfuser_zero";
 
 //! Maximum number of reductions that can be grouped together. The
 //! limit can be increased by extending struct Tuple define in tuple.cu.
-static constexpr int kMaxNumGroupedReductions = 8;
+static constexpr int kMaxNumGroupedReductions = 16;
```
Comment (naoyam) [inline third_party/nvfuser/csrc/type.h:433]: Increased the maximum grouping factor

File: third_party/nvfuser/runtime/tuple.cu [hunk 4]
```diff
@@ -237,6 +237,114 @@ struct Tuple<T0, T1, T2, T3, T4, T5, T6, T7> {
   }
 };
 
+template <
```
Comment (naoyam) [inline third_party/nvfuser/runtime/tuple.cu:240]: Added a 16-way tuple implementation for 16-way grouping.

File: third_party/nvfuser/csrc/grouped_reduction.h [hunk 5]
```diff
@@ -32,8 +32,10 @@ namespace cuda {
 //! welford ops. Unclear how common it would be in practice, though.
 //!
 //! \param reduction_outputs Tensors produced by ReductionOp
```
Comment (zasdfgbnm) [inline third_party/nvfuser/csrc/grouped_reduction.h:34]: nit: Explain what the return value is

**Review by zasdfgbnm — APPROVED** [review]

**Pull request merged**